AwkwardArrowArray.to_pylist() is incompatible with Arrow 20.0.0-rc2

### Version of Awkward Array

2.8.1

### Description and code to reproduce

```
    out = super().to_pylist()
        __class__  = <class 'awkward._connect.pyarrow.extn_types.AwkwardArrowArray'>
        self       = <awkward._connect.pyarrow.extn_types.AwkwardArrowArray object at 0x7fff55a020b0>
[
  [
    1.1,
    2.2,
    3.3
  ],
  [],
  [
    4.4,
    5.5
  ]
]
pyarrow/array.pxi:1775: in pyarrow.lib.Array.to_pylist
    ???
pyarrow/scalar.pxi:1332: in pyarrow.lib.ExtensionScalar.as_py
    ???
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

>   ???
E   TypeError: AwkwardArrowArray.to_pylist() got an unexpected keyword argument 'maps_as_pydicts'
```

The `maps_as_pydicts` parameter was added in https://github.com/apache/arrow/pull/45471.

--- a/src/awkward/_connect/pyarrow/extn_types.py
+++ b/src/awkward/_connect/pyarrow/extn_types.py
@@ -18,7 +18,7 @@ np = NumpyMetadata.instance()
 
 class AwkwardArrowArray(pyarrow.ExtensionArray):
     def to_pylist(self, maps_as_pydicts=None):
-        out = super().to_pylist(maps_as_pydicts=maps_as_pydicts)
+        out = super().to_pylist()
         if (
             isinstance(self.type, AwkwardArrowType)
             and self.type.node_type == "RecordArray"

--- a/src/awkward/_connect/pyarrow/extn_types.py
+++ b/src/awkward/_connect/pyarrow/extn_types.py
@@ -17,8 +17,8 @@ np = NumpyMetadata.instance()
 
 
 class AwkwardArrowArray(pyarrow.ExtensionArray):
-    def to_pylist(self):
-        out = super().to_pylist()
+    def to_pylist(self, maps_as_pydicts=None):
+        out = super().to_pylist(maps_as_pydicts=maps_as_pydicts)
         if (
             isinstance(self.type, AwkwardArrowType)
             and self.type.node_type == "RecordArray"

